apps/obsidian/src/components/DropdownSelect.tsx

```diff
@@ -0,0 +1,85 @@
+import { DropdownComponent } from "obsidian";
+import { useEffect, useRef } from "react";
+
+type DropdownSelectProps<T> = {
+  options: T[];
+  onSelect: (item: T | null) => void;
+  placeholder?: string;
+  getItemText: (item: T) => string;
+};
+
+const DropdownSelect = <T,>({
+  options,
+  onSelect,
+  placeholder = "Select...",
+  getItemText,
+}: DropdownSelectProps<T>) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const dropdownRef = useRef<DropdownComponent | null>(null);
+
+  useEffect(() => {
+    if (!containerRef.current) return;
+
+    if (!dropdownRef.current) {
+      dropdownRef.current = new DropdownComponent(containerRef.current);
+    }
+
+    const dropdown = dropdownRef.current;
+    const currentValue = dropdown.getValue();
+
+    dropdown.selectEl.empty();
+
+    dropdown.addOption("", placeholder);
+
+    options.forEach((option) => {
+      const text = getItemText(option);
+      dropdown.addOption(text, text);
+    });
+
+    if (
+      currentValue &&
+      options.some((opt) => getItemText(opt) === currentValue)
+    ) {
+      dropdown.setValue(currentValue);
+    }
```

Comment:
nit: `getItemText` is running on all `options` three times in this component.  This could be optimized.
Not a big problem for now as we are just doing `option.label`.